src/composables/fungibleTokens.ts

```diff
@@ -48,8 +50,36 @@ const tokenBalances = useStorageRef<ITokenBalance[]>(
   STORAGE_KEYS.fungibleTokenBalances,
 );
 
+/**
+ * List of custom fungible tokens available on user's protocols.
+ * Includes tokens from the user's account that are not on the main list.
+ */
+const customTokensAvailable = computed((): ProtocolRecord<AssetList> => {
+  const uniqueTokens: IToken[] = uniqBy(tokenBalances.value, 'contractId')
+    .map((tokenBalance) => ({
+      contractId: tokenBalance.contractId,
+      protocol: tokenBalance.protocol,
+      name: tokenBalance?.name!,
+      symbol: tokenBalance?.symbol!,
+    }));
+
+  return uniqueTokens.reduce((customTokens, token) => {
+    const { contractId, protocol } = token;
+    if (!customTokens[protocol]) {
+      customTokens[protocol] = {} as AssetList;
+    }
+    if (!tokensAvailable?.value?.[protocol]?.[contractId]) {
+        customTokens[protocol]![contractId] = token;
+    }
+    return customTokens;
+  }, {} as typeof customTokensAvailable.value);
+});
+
 function getProtocolAvailableTokens(protocol: Protocol): AssetList {
-  return tokensAvailable.value[protocol] || {} as AssetList;
+  return {
+    ...customTokensAvailable.value[protocol],
+    ...tokensAvailable.value[protocol],
+  };
```

Comment:
Instead of merging these two arrays each time the function is fired I suggest to:
1. Rename `tokensAvailable` to `defaultTokensAvailable`
2. Rename `customTokensAvailable` to `tokensAvailable` and merge the default ones with the   ones coming from the balances.